examples/rig-kafka-pub-sub-example/example/README.md

```diff
@@ -0,0 +1,260 @@
+
+```yml
+spring:
+  profiles: local
+  kafka:
+    properties:
+      security.protocol: PLAINTEXT
+    consumer:
+      bootstrap-servers: 127.0.0.1:9094
+    producer:
+      bootstrap-servers: 127.0.0.1:9094
+```
+
+### Setup the Routes in RIG
+These routes are set up in the `rig-proxy.json`. It is also possible to set these up by hitting an endpoint. This is not necessary, but is documented below for reference.
+
+
+We set up the routes in RIG by hitting the `/apis` endpoint.
+
+POST
+```http://localhost:7010/v1/apis```
+
+Body
+```JSON
+{
+  "id": "new-service",
+  "name": "new-service",
+  "auth_type": "none",
+  "auth": {
+    "use_header": false,
+    "header_name": "",
+    "use_query": false,
+    "query_name": ""
+  },
+  "versioned": false,
+  "version_data": {
+    "default": {
+      "endpoints": [
+        {
+          "id": "async-publish",
+          "path": "/api/async-publish",
+          "method": "POST",
+          "target": "kafka",
+          "secured": false
+        },
+        {
+          "id": "sync-publish",
+          "path": "/api/sync-publish",
+          "method": "POST",
+          "target": "kafka",
+          "response_from": "kafka",
+          "secured": false
+        }
+      ]
+    }
+  },
+  "proxy": {
+    "use_env": true,
+    "target_url": "KS_HOST",
+    "port": 9092
+  }
+}
+```
+
+Response
+```JSON
+{
+    "message": "ok"
+}
+```
+
+### RIG -> Kafka -> Java -> Kafka -> RIG
+Set up RIG to output events to the commandline.
+
+```bash
+http --stream ':7000/_rig/v1/connection/sse?subscriptions=[{"eventType":"com.example"}]'
+```
+
+*Output*
+```bash
+⚡ http --stream ':7000/_rig/v1/connection/sse?subscriptions=[{"eventType":"com.example"}]'
+HTTP/1.1 200 OK
+access-control-allow-origin: *
+cache-control: no-cache
+content-type: text/event-stream; charset=utf-8
+date: Thu, 29 Aug 2019 21:31:29 GMT
+server: Cowboy
+transfer-encoding: chunked
+
+event: rig.connection.create
+data: {"data":{"connection_token":"g2dkAA1yaWdAMTI3LjAuMC4xAAAPJwAAAAAD"},"id":"632f6767-0c2e-4166-82ae-27c2be34df43","source":"rig","specversion":"0.2","time":"2019-08-29T21:31:30.362896+00:00","type":"rig.connection.create"}
+
+event: rig.subscriptions_set
+data: {"data":[{"eventType":"com.example","oneOf":[]}],"id":"9a4187a0-af79-439e-b07b-27e98bee9e7f","source":"rig","specversion":"0.2","time":"2019-08-29T21:31:30.369058+00:00","type":"rig.subscriptions_set"}
+
+: heartbeat
+```
+
+Have Kafka output to the CLI for both topics RIG and the Springboot app consume and produce.
+
+```bash
+docker exec -it kafka bash
+```
+
+You now have a command prompt in Kafka.
+
+```bash
+kafka-console-consumer --bootstrap-server kafka:9092 --topic rig --from-beginning
+```
+
+Open a second window
+```bash
+docker exec -it kafka bash
+```
+
+```bash
+kafka-console-consumer --bootstrap-server kafka:9092 --topic rig-consumed --from-beginning
+```
+
+Send a POST request to RIG
+
+```
+http://localhost:7000/api/async-publish
+```
+
+Body:
+```JSON
+{
```

Comment:
Can you please update the formatting here? Looks a bit scuffed and hard to read.